chainer/link.py

```diff
@@ -245,30 +245,35 @@ def copy(self):
             d[name].grad = None
         return ret
 
-    def to_cpu(self):
+    def to_cpu(self, copy=False):
         """Copies parameter variables and persistent values to CPU.
 
         This method does not handle non-registered attributes. If some of such
         attributes must be copied to CPU, the link implementation must
         override this method to do so.
 
-        Returns: self
+        Returns: self if `copy`` is False, otherwise a copied link
```

Comment:
From a user's perspective, I wonder if this API change might cause the following confusion: Without looking at the code, I might look at this documentation and wonder when is it appropriate to use`copy=True`? That is, "When or why should I use this new feature?" I might also wonder what is the difference between first copying a link and then calling `to_cpu()` versus using `copy=True`. The same thought applies to `to_gpu()`. 

Maybe the documentation is fine as is, but if you can think of a way to simply state the motivation for being able to return a copy and when or how it should be used, that might avoid any confusion to the user.